connector/src/main/java/org/geysermc/connector/entity/living/monster/raid/PillagerEntity.java

```diff
@@ -38,12 +38,30 @@ public PillagerEntity(long entityId, long geyserId, EntityType entityType, Vecto
     }
 
     @Override
-    public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession session) {
-        if (entityMetadata.getId() == 16) {
-            // Java Edition always has the Pillager entity as positioning the crossbow
-            metadata.getFlags().setFlag(EntityFlag.USING_ITEM, true);
-            metadata.getFlags().setFlag(EntityFlag.CHARGED, true);
+    public void updateMainHand(GeyserSession session) {
+        checkForCrossbow(session);
+
+        super.updateMainHand(session);
+    }
+
+    @Override
+    public void updateOffHand(GeyserSession session) {
+        checkForCrossbow(session);
+
+        super.updateOffHand(session);
+    }
+
+    /**
+     * Check for a crossbow in either the mainhand or offhand. If one exists, indicate that the pillager should be posing
+     */
+    protected void checkForCrossbow(GeyserSession session) {
```

Comment:
Does this not affect skeletons? Also, does this also happen with bows on them? (maybe out of scope for the PR, but then again this fix is kinda already out of scope)

Reply:
https://github.com/GeyserMC/Geyser/blob/master/connector/src/main/java/org/geysermc/connector/entity/living/monster/AbstractSkeletonEntity.java#45 no - their hands raised when "angered".